eclipse-collections/src/main/java/org/eclipse/collections/impl/map/immutable/AbstractImmutableMap.java

```diff
@@ -201,6 +202,9 @@ public void clear()
         return map.toImmutable();
     }
 
+    @Override
+    public abstract AbstractImmutableSet<K> keySet();
```

Comment:
@johnjstuart what's the point of specifying that it should implement `AbstractImmutableSet`? I wouldn't specify anything in this class but only in subclasses IMHO.